core/src/main/java/com/linecorp/armeria/common/stream/DeferredStreamMessage.java

```diff
@@ -184,33 +176,47 @@ public void subscribe(Subscriber<? super T> subscriber, Executor executor, boole
     }
 
     private void subscribe0(Subscriber<? super T> subscriber, Executor executor, boolean withPooledObjects) {
+        if (abortPending) {
+            failLateSubscriber(subscriber, executor, AbortedStreamException.get());
+            return;
+        }
+
         final PendingSubscription<? super T> newPendingSubscription =
                 new PendingSubscription<>(subscriber, executor);
 
-        if (!pendingSubscriptionUpdater.compareAndSet(this, null, newPendingSubscription)) {
-            PendingSubscription<? super T> oldPendingSubscription = pendingSubscription;
-            if (oldPendingSubscription == ABORTED_SUBSCRIPTION) {
-                throw new IllegalStateException("cannot subscribe to an aborted publisher");
-            } else {
-                throw new IllegalStateException("subscribed by other subscriber already: " +
-                                                oldPendingSubscription.subscriber);
+        final StreamMessage<T> delegate = this.delegate;
+        if (delegate == null) {
```

Comment:
What if the delegate is not null and subscribed by other subscriber? let the delegate throw the exception?